app.rb

```diff
@@ -1,29 +1,50 @@
 require 'sinatra/base'
-require './lib/rps_game.rb'
-
+require './lib/rps_game'
+require './lib/computer'
 
 class RockPaperScissors < Sinatra::Base
+  enable :sessions
+
   get '/test' do
     'test page'
   end
 
-
-  get '/enter-name' do
-    @name = params[:name]
-    erb :index
+  get '/start-game' do
+    erb :game_start
   end
 
-  get '/weapon-choice' do
-    @rock = params[:rock]
-    @paper = params[:paper]
-    @scissors = params[:scissors]
-    erb :weapon
+  post '/name' do
+    session[:name] = params[:name]
+    redirect '/fight'
   end
 
   get '/fight' do
+    @name = session[:name]
     erb :fight
   end
 
+  get '/rock' do
```

Comment:
I'm wondering whether the reason Sinatra is failing here is because you have a post syntax in your erb file but a get syntax in your controller.